runtime-decompiler/src/main/java/org/jrd/backend/data/Cli.java

```diff
@@ -613,36 +613,42 @@ private void listJvms(List<String> args) {
+                "\t" + LISTJVMS + "\t\t\t\t\t\t\t\t\t\t\tList all local Java processes and their PIDs.\n" +
+                "\t" + LISTPLUGINS + "\t\t\t\t\t\t\t\t\t\tList all currently configured decompiler plugins and their statuses.\n" +
+                "\t" + LISTCLASSES + " <PUC> [<CLASS REGEX>...]\t\t\t\tList all loaded classes of a process, optionally filtering them.\n" +
+                "\t" + COMPILE + " [-p <PLUGIN>] [-cp <PUC>] [-r] <PATH>...\tCompile local files against runtime classpath, specified by -cp.\n" +
+                "\t\t\t\t\t\t\t\t\t\t\t\t\t\tUse -p to utilize some plugins' (like jasm or jcoder) bundled compilers.\n" +
+                "\t\t\t\t\t\t\t\t\t\t\t\t\t\tUse -r for recursive search if <PATH> is a directory.\n" +
+                "\t\t\t\t\t\t\t\t\t\t\t\t\t\tIf the argument of " + SAVEAS + " is a valid PID or URL, the compiled code will be attempted to be injected into that process.\n" +
+                "\t\t\t\t\t\t\t\t\t\t\t\t\t\tIf multiple PATHs were specified, but no " + SAVEAS + ", the process fails.\n" +
+                "\t" + DECOMPILE + " <PUC> <PLUGIN> <CLASS REGEX>...\t\t\tDecompile and print classes of a process with the specified decompiler plugin.\n" +
+                "\t\t\t\t\t\t\t\t\t\t\t\t\t\tJavap can be passed options by appending them without spaces: 'javap-v-public ...' executes as 'javap -v -public ...'\n" +
+                "\t" + OVERWRITE + " <PUC> <CLASS NAME> [<CLASS FILE>]\t\tOverwrite class of a process with new bytecode. If <CLASS FILE> is not set, standard input is used.\n" +
+                "Saving modifiers:\n" +
+                "\t" + SAVEAS + " <PATH>\t\t\t\tAll outputs will be written here instead of to standard output.\n" +
+                "\t" + SAVELIKE + " <SAVE METHOD>\t\tSpecify how saving will behave.\n" +
+                "Additional information:\n" +
+                "\tAll options can be with either one or two leading slashes ('-').\n" +
+                "\tWhen using <CLASS REGEX>, don't forget to escape dollar signs '$' of inner classes to '\\$' , as otherwise they are treated as end-of-line by REGEX.\n" +
+                "\t<PUC>, or PidUrlClasspath, can be one of:\n" +
+                "\t\t- local process PID\n" +
+                "\t\t- remote process URL, in the format of 'hostname:port'\n" +
+                "\t\t- classpath of JAR on the filesystem (classpath separator is '" + File.pathSeparator + "')\n" +
+                "\t<SAVE METHOD> can be one of:\n" +
+                "\t\t'" + Saving.DIR + "' - Result will be saved as '<PATH>/fully/qualified/name.class'. Default for .class binaries.\n" +
+                "\t\t'" + Saving.FQN + "' - Result will be saved as '<PATH>/fully.qualified.name.java'. Default for .java sources.\n" +
+                "\t\t'" + Saving.EXACT + "' - Result will be saved exactly to '<PATH>'. Default for everything else.\n";
+
+        System.out.print(helpString);
     }
```

Comment:
oh god :D waht was bad with spaces? I might have issues with tabs - as each terminalhave different size of tab, it maygo very wild...

Reply:
Maybe decalre TAB constant of "    ", and use that? At elast it vill be versatiel enough to repalce it by \t x spaces as needed. Hoepfully nIkol will succeed with repalcing it by beust jcommander.

Reply:
Fixed it with spaces in 91eac3b. Looks garbled in code, but the output is pretty cool:
```
Usage:
  (./start.sh|start.bat)  # launches GUI
  (./start.sh|start.bat) [-verbose] (-help|-listjvms|-listplugins|-overwrite)
  (./start.sh|start.bat) [-verbose] (-listclasses|-bytes|-base64bytes|-compile|-decompile) [-saveas <DIR> [-savelike <SAVE METHOD>]]
Available options:
  -help, -h                                         Print this help text.
  -verbose                                          All exceptions and some debugging strings will be printed to standard error.
  -base64bytes <PUC> <CLASS REGEX>...               Print Base64 encoded binary form of requested classes of a process.
  -bytes <PUC> <CLASS REGEX>...                     Print binary form of requested classes of a process.
  -listjvms                                         List all local Java processes and their PIDs.
  -listplugins                                      List all currently configured decompiler plugins and their statuses.
  -listclasses <PUC> [<CLASS REGEX>...]             List all loaded classes of a process, optionally filtering them.
  -compile [-p <PLUGIN>] [-cp <PUC>] [-r] <PATH>... Compile local files against runtime classpath, specified by -cp.
                                                    Use -p to utilize some plugins' (like jasm or jcoder) bundled compilers.
                                                    Use -r for recursive search if <PATH> is a directory.
                                                    If the argument of -saveas is a valid PID or URL, the compiled code will be attempted to be injected into that process.
                                                    If multiple PATHs were specified, but no -saveas, the process fails.
  -decompile <PUC> <PLUGIN> <CLASS REGEX>...        Decompile and print classes of a process with the specified decompiler plugin.
                                                    Javap can be passed options by appending them without spaces: 'javap-v-public ...' executes as 'javap -v -public ...'
  -overwrite <PUC> <CLASS NAME> [<CLASS FILE>]      Overwrite class of a process with new bytecode. If <CLASS FILE> is not set, standard input is used.
Saving modifiers:
  -saveas <PATH>                                    All outputs will be written to PATH instead of to standard output.
  -savelike <SAVE METHOD>                           Specify how saving will behave.
Additional information:
  All options can be with either one or two leading slashes ('-').
  When using <CLASS REGEX>, don't forget to escape dollar signs '$' of inner classes to '\$' , as otherwise they are treated as end-of-line by REGEX.
  <PUC>, or PidUrlClasspath, can be one of:
      - local process PID
      - remote process URL, in the format of 'hostname:port'
      - classpath of JAR on the filesystem (classpath separator is ';')
  <SAVE METHOD> can be one of:
      'dir' - Result will be saved as '<PATH>/fully/qualified/name.class'. Default for .class binaries.
      'fqn' - Result will be saved as '<PATH>/fully.qualified.name.java'. Default for .java sources.
      'exact' - Result will be saved exactly to '<PATH>'. Default for everything else.
```
@judovana got any issues regarding the content? I tried my best to de-vanek the text while also not leaving anything out :) Nikol can use/be inspired by this text when filling out the annotations for JCommander.